packages/wallet/src/cip30.ts

```diff
@@ -257,11 +257,11 @@ export const createWalletApi = (
   signTx: async (tx: Cbor, _partialSign?: Boolean): Promise<Cbor> => {
     const scope = new ManagedFreeableScope();
     logger.debug('signTx');
-    const txDecoded = scope.manage(CML.TransactionBody.from_bytes(Buffer.from(tx, 'hex')));
+    const txDecoded = scope.manage(CML.Transaction.from_bytes(Buffer.from(tx, 'hex')));
     const hash = Cardano.TransactionId(
-      Buffer.from(scope.manage(CML.hash_transaction(txDecoded)).to_bytes()).toString('hex')
+      Buffer.from(scope.manage(CML.hash_transaction(txDecoded.body())).to_bytes()).toString('hex')
```

Comment:
This clones the tx body, so need to free() afterwards too
```suggestion
      Buffer.from(scope.manage(CML.hash_transaction(scope.manage(txDecoded.body()))).to_bytes()).toString('hex')
```